### PR TITLE
Transition guide

### PR DESCRIPTION
**Pull request by mshabunin**
This document describes some aspects of 2.4->3.0 transition process.

### CODE REVIEW

File: doc/tutorials/introduction/transition_guide/transition_guide.markdown [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+This is a place for all new, experimental and non-free algorithms. It does not receive so much attention from the support team comparing to main repository, but the community makes an effort to keep it in a good shape.
+
+To build OpenCV with _contrib_ repository, add the following option to your cmake command: `-DOPENCV_EXTRA_MODULES_PATH=<path-to_opencv_contrib>/modules`
+
+##### Headers layout
+Old headers were like `opencv2/<module>/<module>.hpp` and now there are top-level module headers containing the most of the functionality: `opencv2/<module>.hpp`. All C-style API were moved to separate headers like `opencv2/core/core_c.h`.
+
+##### Algorithm interfaces
+General algorithm usage pattern has changed: now it must be created on heap wrapped in smart pointer `cv::Ptr`. Version 2.4 allowed both stack and heap allocations, directly or via smart pointer.
```
Comment (StevenPuttemans) [inline doc/tutorials/introduction/transition_guide/transition_guide.markdown]: I would add an explicit sample here. The Q&A has been flooded with questions about this!

**Comment by ilya-lavrenov** [comment]
:+1:

**Comment by PhilLab** [comment]
Just out of curiosity: where can I find the compiled tutorial in the trunk?
Its [not listed](http://docs.opencv.org/trunk/doc/tutorials/introduction/table_of_content_introduction/table_of_content_introduction.html) and the [search engine](http://docs.opencv.org/trunk/search.html?q=transition&check_keywords=yes&area=default) also does not find it. And it is the same with several other tutorials (e.g. the new documentation tutorial).

**Comment by mshabunin** [comment]
@PhilLab, you can find it here: http://docs.opencv.org/ref/master/ ([documentation tutorial](http://docs.opencv.org/ref/master/d4/db1/tutorial_documentation.html), [transition guide](http://docs.opencv.org/ref/master/db/dfa/tutorial_transition_guide.html))
Currently nightly docs for master are uploaded to `/ref/master`, I'll change this path to `/master` (`/trunk`) by the end of this week.

**Comment by PhilLab** [comment]
thanks

**Pull request merged**